src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseService.java

```diff
@@ -93,29 +92,28 @@ public Optional<String> findCaseId(String taskId) {
         return Optional.empty();
     }
 
-    public GetTasksResponse<Task> searchForTasks(SearchRequest searchRequest,
+    public GetTasksResponse<Task> searchForTasks(int firstResult,
+                                                 int maxResults,
+                                                 SearchRequest searchRequest,
                                                  AccessControlResponse accessControlResponse,
                                                  boolean granularPermissionResponseFeature) {
 
         List<RoleAssignment> roleAssignments = accessControlResponse.getRoleAssignments();
```

Comment:
I think it's a good idea to make sure we keep the original list of roleAssignments received from AM vs the filtered roleAssignments. This is for debugging or if in future it is needed in a different context.

Have to be a bit careful about copies in that it is done in a thread safe way.

Reply:
Done in RWA-2212 pr